modules/fscloud/variables.tf

```diff
@@ -69,74 +69,15 @@ variable "allow_is_to_cos" {
-    databases-for-elasticsearch = optional(string)
-    databases-for-enterprisedb  = optional(string)
-    databases-for-etcd          = optional(string)
-    databases-for-mongodb       = optional(string)
-    databases-for-mysql         = optional(string)
-    databases-for-postgresql    = optional(string)
-    databases-for-redis         = optional(string)
-    directlink                  = optional(string)
-    iam-groups                  = optional(string)
-    is                          = optional(string)
-    messagehub                  = optional(string)
-    messages-for-rabbitmq       = optional(string)
-    schematics                  = optional(string)
-    secrets-manager             = optional(string)
-    server-protect              = optional(string)
-    user-management             = optional(string)
-    apprapp                     = optional(string)
-    compliance                  = optional(string)
-    event-notifications         = optional(string)
-    logdna                      = optional(string)
-    logdnaat                    = optional(string)
-    cloudantnosqldb             = optional(string)
-    globalcatalog-collection    = optional(string)
-    sysdig-monitor              = optional(string)
-    sysdig-secure               = optional(string)
-    toolchain                   = optional(string)
-  })
-  default = {
-    cloud-object-storage        = null
-    codeengine                  = null
-    containers-kubernetes       = null
-    databases-for-cassandra     = null
-    databases-for-elasticsearch = null
-    databases-for-enterprisedb  = null
-    databases-for-etcd          = null
-    databases-for-mongodb       = null
-    databases-for-mysql         = null
-    databases-for-postgresql    = null
-    databases-for-redis         = null
-    directlink                  = null
-    iam-groups                  = null
-    is                          = null
-    messagehub                  = null
-    messages-for-rabbitmq       = null
-    schematics                  = null
-    secrets-manager             = null
-    server-protect              = null
-    user-management             = null
-    apprapp                     = null
-    compliance                  = null
-    event-notifications         = null
-    logdna                      = null
-    logdnaat                    = null
-    cloudantnosqldb             = null
-    globalcatalog-collection    = null
-    sysdig-monitor              = null
-    sysdig-secure               = null
-    toolchain                   = null
-  }
+variable "zone_service_ref_map" {
+  type = map(object({
+    zone_name           = optional(string)
+    serviceRef_location = optional(list(string))
+  }))
+  default = {}
   validation {
     condition = alltrue([
-      for service_ref, service_ref_name in var.zone_service_ref_list : contains([
+      for service_ref, service_ref_name in var.zone_service_ref_map : contains([
```

Comment:
second argument here is an object rather than the service_ref_name

Reply:
Modified to use only map keys to validate.